providers/domain.rb

```diff
@@ -38,17 +38,22 @@
       cmd << " -DomainName #{new_resource.name}"
       cmd << " -SafeModeAdministratorPassword (convertto-securestring '#{new_resource.safe_mode_pass}' -asplaintext -Force)"
       cmd << ' -Force:$true'
+      cmd << ' -NoRebootOnCompletion' if !new_resource.restart
     else node[:os_version] <= '6.1'
       cmd = 'dcpromo -unattend'
       cmd << " -newDomain:#{new_resource.type}"
       cmd << " -NewDomainDNSName:#{new_resource.name}"
-      cmd << ' -RebootOnCompletion:Yes'
+      if !new_resource.restart
```

Comment:
How about something like this? Not sure if Ruby ternaries will work this way. (Not much of a rubyist yet, mind you...)
`new_resource.restart ? cmd << ' -RebootOnCompletion:Yes' : cmd << ' -RebootOnCompletion:No'`